### ISSUE TITLE
Scrolling in browser does not work correctly when Avalonia content does not start at the top-left corner

### ISSUE THREAD

**Issue by koustubhmoharir** [AvaloniaUI/Avalonia#14416]
## Describe the bug
When Avalonia content is rendered in a part of the page instead of the full page, scrolling within a ScrollViewer does not work correctly. It appears that the mouse coordinates are being considered relative to the page instead of relative to the top left corner of Avalonia content. There is a "dead area" at the right and bottom of the scrollviewer whose width and height match the offset of the top left corner of the Avalonia content w.r.t the page.

## To Reproduce

1. Create an Avalonia browser application targeting .NET 8 with Avalonia 11.0.7
2. Replace the contents of index.html with 
```
<!DOCTYPE html>
<html style="height:100%;">

<head>
    <title>AvaTestNet8.Browser</title>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <base href="/" />
    <link rel="modulepreload" href="./main.js" />
    <link rel="modulepreload" href="./dotnet.js" />
    <link rel="modulepreload" href="./avalonia.js" />
    <link rel="stylesheet" href="./app.css" />
</head>

<body style="margin: 0px; overflow: hidden;display:flex;flex-direction:column; height:100%;align-items:stretch;">
    <div id="out" style="position:relative;padding:0;outline:none;height:400px;width:400px;margin:100px;">
        <div>Loading...</div>
    </div>
    <script type='module' src="./main.js"></script>
</body>

</html>
```
3. Replace the content in MainView.xaml with
```
<ScrollViewer HorizontalScrollBarVisibility="Auto" VerticalScrollBarVisibility="Auto">
  <Rectangle Height="4000" Width="4000">
    <Rectangle.Fill>
      <LinearGradientBrush>
        <LinearGradientBrush.GradientStops>
          <GradientStop Color="Yellow" Offset="0"/>
          <GradientStop Color="Green" Offset="1"/>
        </LinearGradientBrush.GradientStops>
      </LinearGradientBrush>
    </Rectangle.Fill>
  </Rectangle>
</ScrollViewer>
```
5. Run the application in the browser and try to scroll with the mouse wheel when the cursor is near the bottom or right edges of the Avalonia content. Scrolling works around the top edges of the content but not near the bottom edges. The width / height of the "dead areas" match the offset of the content.

## Expected behavior
Scrolling should work correctly.

## Environment 

- OS: Browser
- Avalonia-Version: 11.0.7

**Comment by stevemonaco** [comment]
Not commenting on if this is a bug or by-design.

I've been able to use `iframe`s instead of `div` to create a page with multiple Avalonia WASM views side-by-side and the mouse inputs were correct. You may be able to do the same, but my scenario was a bit different. This was a separate webpage that embedded already web-deployed WASM apps as a test (the Solitaire demo). Not sure how to integrate `iframe` into the WASM app itself for mixed content.

**Comment by jp2masa** [comment]
Duplicate of #12780?

**Comment by maxkatz6** [comment]
Yeah, it’s not a bug. Only a confusion with how CSS works.

**Comment by koustubhmoharir** [comment]
Please note that the "out" div does have position relative. Also, click events and all other mouse interactions work correctly.

Looking at the source code in AvaloniaView.cs, I see that the OnWheel handler uses clientX and clientY. Other handlers use offsetX and offsetY.
https://github.com/AvaloniaUI/Avalonia/blob/2a9e7e9dcfc3e590bdd50b1e874ce70c0fafcc24/src/Browser/Avalonia.Browser/AvaloniaView.cs#L259

**Comment by koustubhmoharir** [comment]
It looks like a very safe fix because OnWheel is the only handler that uses clientX and clientY. All other handlers use a utility method ExtractRawPointerFromJSArgs which uses offsetX and offsetY.

I will be able to open a PR only after a few weeks since this is not a showstopper for me right now.

**Comment by maxkatz6** [comment]
Haven't paid attention that issue is mouse wheel specific. Yes, changing to offsetX/offsetY makes sense.

**Comment by timunie** [comment]
@koustubhmoharir a PR would be very welcome
